src/odemis/acq/stitching/_registrar.py

```diff
@@ -124,100 +127,105 @@ def addTile(self, tile, dependent_tiles=None):
-            shift = self.registered_positions[ti[0]][ti[1]]
-            tile_positions.append(((shift[0] + firstPosition[0]) * self.px_size[0],
-                                   (firstPosition[1] - shift[1]) * self.px_size[1]))
+        for ti in self.acq_order:
+            shift = self.registered_positions_px[ti[0]][ti[1]]
+            tile_positions.append((first_position_m[0] + shift[0] * self.px_size[0],
+                                   first_position_m[1] - shift[1] * self.px_size[1]))
 
         # Return positions for dependent tiles
         for t, sdts in zip(tile_positions, self.shift_tile_dep_tiles):
             dts = []  # dependent tiles for tile
             for sdt in sdts:
                 dts.append((t[0] + sdt[0], t[1] + sdt[1]))
             dep_tile_positions.append(dts)
+        self.registered_positions_m = tile_positions
+        return self.registered_positions_m, dep_tile_positions
+
+    def _insert_tile_to_grid(self, tile):
+        """
+        Stores the tile at the proper place in the grid. If necessary, the grid is
+        extended to accommodate the new tile.
+
+        :param tile: (DataArray) tile to be inserted
+        :returns: (int, int) row, col grid position of the tile
+        :updates self.tiles, self.posX, self.posY, self.ovrlp, self.px_size, self.size:
+        """
+        # Find position of the tile in the grid. Indices of grid position are
+        # given as posX, posY.
+        if self.tiles[0][0] is None:
+            self.size = tile.shape
+            self.px_size = tile.metadata[model.MD_PIXEL_SIZE]
+            self.pos_x = 0
+            self.pos_y = 0
+
+            self.pos_prev_x = 0  # used to determine if tile should be compared to right or left tile
+            return
+
+        if tile.shape != self.tiles[0][0].shape:
+            raise ValueError("Tile shape differs from previous tile shapes %s != %s" %
+                             (tile.shape, self.tiles[0][0].shape))
+
+        md_pos = tile.metadata[model.MD_POS]
+
+        # Find the registered tile that is closest to the new tile.
+        pos_prev, md_pos_prev = self._find_closest_tile(md_pos)
+
+        # Insert new tile either to the right or to the bottom of the
+        # closest tile.
+        ver_diff = md_pos[1] - md_pos_prev[1]
+        hor_diff = md_pos[0] - md_pos_prev[0]
+        if abs(ver_diff) > abs(hor_diff):
+            if ver_diff < 0:
+                self.pos_y = pos_prev[0] + 1
+                self.pos_x = pos_prev[1]
+                if len(self.registered_positions_px) <= self.pos_y:
+                    self._updateGrid("y")  # extend grid in y direction
+            else:
+                self.pos_y = pos_prev[0] - 1
+                self.pos_x = pos_prev[1]
+        else:
+            if hor_diff > 0:
+                self.pos_y = pos_prev[0]
+                self.pos_x = pos_prev[1] + 1
+                if len(self.registered_positions_px[0]) <= self.pos_x:
```

Comment:
This check is a bit different from line 201?

Reply:
That is correct. It is a list of lists, therefore `len(self.registered_positions_px)` gives the amount of vertical tiles, and `len(self.registered_positions_px[0])` gives the number of horizontal tiles in the first line